mypy/plugins/dataclasses.py

```diff
@@ -333,6 +333,38 @@ def collect_attributes(self) -> Optional[List[DataclassAttribute]]:
 
             node = sym.node
             assert not isinstance(node, PlaceholderNode)
+
+            if isinstance(node, TypeAlias):
+                ctx.api.fail(
+                    (
+                        'Type aliases inside dataclass definitions '
+                        'are not supported at runtime.'
+                    ),
+                    Context(line=node.line, column=node.column)
```

Comment:
It should be enough to pass `node` here, without constructing a `Context` instance.